Security considerations for client_metadata parameters (#735)
openid/OpenID4VP: 1.0/openid-4-verifiable-presentations-1_0.md

MUST perform its own security checks on the returned Credentials and
Presentations.

## Source of Verifier Metadata

> Wallet implementers need to choose a source of client metadata they can trust within their own ecosystem or implementation. This should determine whether it is safe to use or display to a User. For example, a client name obtained from an X.509 leaf certificate that chains to a trust anchor the Wallet trusts may be appropriate to be displayed to an End-User, whereas a client name from the `client_metadata` parameter is not suitable to display. 

Reviewer: This uses a "should" which we should avoid here I think.
It's not really clear what the second sentence actually is intended to mean: "should" it determine or does it determine?

Reviewer 2: This addresses both your points I think, as well as cleaning up a bit of terminology and simplifying wording:

```suggestion
Wallet implementers need to choose a source of Verifier metadata they can trust within their own ecosystem or implementation, including determining whether metadata is safe to use or display to an End-User. For example, a Verifier name obtained from an X.509 leaf certificate that chains to a trust anchor the Wallet trusts may be suitable to display to an End-User, whereas a Verifier name from the `client_metadata` parameter is not suitable to display.
```

> It is recommended to use `client_metadata` parameters only for client-supplied parameters that can be used untrusted, for example, a public key used to encrypt the response. 

Reviewer: This uses a "recommended" which we should probably either avoid or capitalize.

Also, the example should be a new sentence.

Reviewer 2: Applying Frederik's suggestion, plus also resolving my currently unresolved comment https://github.com/openid/OpenID4VP/pull/735#discussion_r3528561569

```suggestion
Profiles of this specification MAY define additional `client_metadata` parameters, as described in (#new_parameters). It is RECOMMENDED that such parameters are limited to client-supplied values that can safely be used without being trusted. One example of such a value is a public key used to encrypt the Authorization Response.
```

> Ecosystems can use other ecosystem-specific trusted data sources as long as they conduct a proper trust evaluation. 

Reviewer: can -> MAY?

Reviewer 2: Suggestion that resolves that and my own ecosystem vs profiles point.

```suggestion
Wallets MAY use other ecosystem-specific trusted sources of Verifier metadata, provided the Wallet is able to establish trust in those sources. As described in (#new_parameters), authoritative data obtained from such sources takes precedence over the values passed in the `client_metadata` parameter.
```
